src/util/train.hpp

```diff
@@ -0,0 +1,80 @@
+#pragma once
+#include "gta/enums.hpp"
+#include "services/players/player_service.hpp"
+#include "script_global.hpp"
+#include "pointers.hpp"
+#include "natives.hpp"
+#include "script.hpp"
+
+namespace big::train
+{
+	inline auto get_all_vehicles()
+	{
+		std::vector<Vehicle> result;
+		rage::CReplayInterface* CReplayInterface_var = *g_pointers->m_replay_interface; 
+		for (int i = 0; i <= 300; i++)
+		{
+			auto vehicle_ptr = CReplayInterface_var->m_vehicle_interface->get_vehicle(i);
+			if (vehicle_ptr)
+			{
+				Vehicle vehicle_handle = g_pointers->m_ptr_to_handle(vehicle_ptr);
+
+				result.push_back(vehicle_handle);
+			}
+		}
+			return result;
+	};
+
+	inline int get_closest_train()
+	{
+		auto allVehicles = get_all_vehicles();
+
+		for (int i = 0; i < allVehicles.size(); i++)
+		{
+			if (ENTITY::GET_ENTITY_MODEL(allVehicles[i]) == 1030400667)
+				return allVehicles[i];
+		}
+		return 0;
+	}
+
+	inline void hijack_train()
+	{
+		auto train = get_closest_train();
+
+		if (train != 0)
+		{
+			NETWORK::NETWORK_REQUEST_CONTROL_OF_ENTITY(train);
+
+			while (!NETWORK::NETWORK_HAS_CONTROL_OF_ENTITY(train))
+			{
+				script::get_current()->yield();
+			}
```

Comment:
Could end up infinitely looping causing the fiber pool to lockup, make use of the `entity::request_control_of()` function which has a built in breakout if it fails to get control of the entity.

Reply:
assuming you meant entity::take_control_of, switched it over

Reply:
Yup, seems correct now.